examples/legacy/tutorial_5_neuromorphic_datasets.ipynb

```diff
@@ -60,89 +86,323 @@
+    "                                     ])\n",
     "\n",
-    "download = True\n",
-    "trainset = tonic.datasets.NMNIST(save_to='./data', download=download, transform=transform, train=True)\n",
-    "testset = tonic.datasets.NMNIST(save_to='./data', download=download, transform=transform, train=False)"
+    "trainset = tonic.datasets.NMNIST(save_to='./data', transform=frame_transform, train=True)\n",
+    "testset = tonic.datasets.NMNIST(save_to='./data', transform=frame_transform, train=False)"
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "3831428d-0511-4fde-84d9-11d08fa45df7",
+   "id": "70be0b77-9405-44f6-af49-dfc4d49566c6",
    "metadata": {},
    "source": [
-    "Now we can start loading data much like we would load from any other Pytorch dataloader. Because SNNTorch uses time as its first dimension, we have to permute the tensor dimensions from (Batch, Time, Channels, Width, Height) to (Time, Batch, Channels, Width, Height). "
+    "The original data is stored in a format that is slow to read. To speed up dataloading, we can make use of disk caching. That means that once files are loaded from the original file, they are written to disk in an efficient format in our cache directory. Let's compare some file reading speeds to read 100 examples."
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": null,
-   "id": "5b35c7cd-d292-47cd-9203-7f31aa7f7207",
+   "execution_count": 5,
+   "id": "3a6bf2a2-ff9f-4cdc-8cb3-02a1a9d71a11",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def load_sample_simple():\n",
+    "    for i in range(100):\n",
+    "        events, target = trainset[i]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "1a9d3b28-b303-4a17-be78-b9918911a7cd",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "1.68 s ± 7.76 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "<TimeitResult : 1.68 s ± 7.76 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)>"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "%timeit -o load_sample_simple()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b957b32f-d76b-42c0-8c1c-b6b63e84e2ef",
+   "metadata": {},
+   "source": [
+    "Let's try to decrease the time to read 100 samples by using a PyTorch DataLoader as well as disk caching."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "f47e0798-5259-491a-9d3a-59b13b1b0983",
    "metadata": {},
    "outputs": [],
    "source": [
     "from torch.utils.data import DataLoader\n",
-    "trainloader = DataLoader(trainset, shuffle=True)\n",
+    "from tonic import CachedDataset\n",
+    "\n",
+    "cached_trainset = CachedDataset(trainset, cache_path='./cache/nmnist/train')\n",
```

Comment:
When running in colab, the `cache_path` is created, but the hdf5 file is not. I get the following error (same issue in the `fast_dataloading` tutorial in the Tonic docs):

> OSError: Unable to open file (unable to open file: name = 'data/cache/nmnist/train/0_0.hdf5', errno = 2, error message = 'No such file or directory', flags = 0, o_flags = 0)

Reply:
fixed it in version 1.0.4
the issue was an older version of h5py